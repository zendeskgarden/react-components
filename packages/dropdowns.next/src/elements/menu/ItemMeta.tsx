/**
 * Copyright Zendesk, Inc.
 *
 * Use of this source code is governed under the Apache License, Version 2.0
 * found at http://www.apache.org/licenses/LICENSE-2.0.
 */

import React, { HTMLAttributes, forwardRef } from 'react';
import useItemContext from '../../context/useItemContext';
import { StyledItemMeta } from '../../views';

const ItemMetaComponent = forwardRef<HTMLDivElement, HTMLAttributes<HTMLDivElement>>(
  (props, ref) => {
    const { isDisabled } = useItemContext();

    return <StyledItemMeta isDisabled={isDisabled} {...props} ref={ref} />;
  }
);

ItemMetaComponent.displayName = 'Item.Meta';

/**
 * @extends HTMLAttributes<HTMLDivElement>
 */
export const ItemMeta = ItemMetaComponent;
