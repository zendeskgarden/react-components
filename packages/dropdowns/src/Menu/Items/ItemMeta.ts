/**
 * Copyright Zendesk, Inc.
 *
 * Use of this source code is governed under the Apache License, Version 2.0
 * found at http://www.apache.org/licenses/LICENSE-2.0.
 */

import { StyledItemMeta } from '../../styled';

/**
 * Accepts all `<div>` props
 */
const ItemMeta = StyledItemMeta;

ItemMeta.displayName = 'ItemMeta';

/** @component */
export default ItemMeta;
