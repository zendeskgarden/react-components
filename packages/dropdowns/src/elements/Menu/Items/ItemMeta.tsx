/**
 * Copyright Zendesk, Inc.
 *
 * Use of this source code is governed under the Apache License, Version 2.0
 * found at http://www.apache.org/licenses/LICENSE-2.0.
 */

import React, { HTMLAttributes } from 'react';
import { StyledItemMeta } from '../../../styled';
import useMenuContext from '../../../utils/useMenuContext';
import useItemContext from '../../../utils/useItemContext';

/**
 * Accepts all `<div>` props
 */
export const ItemMeta = React.forwardRef<HTMLSpanElement, HTMLAttributes<HTMLSpanElement>>(
  (props, ref) => {
    const { isCompact } = useMenuContext();
    const { isDisabled } = useItemContext();

    return <StyledItemMeta ref={ref} isCompact={isCompact} isDisabled={isDisabled} {...props} />;
  }
);

ItemMeta.displayName = 'ItemMeta';
