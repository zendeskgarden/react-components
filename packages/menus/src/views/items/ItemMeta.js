import styled from 'styled-components';
import { retrieveTheme } from '@zendesk/garden-react-theming';
import MenuStyles from '@zendesk/garden-css-menus';

import { version } from '../../../package.json';
const COMPONENT_ID = 'menus.item_meta';

/**
 * Accepts all `<span>` props
 */
const ItemMeta = styled.span.attrs({
  'data-garden-id': COMPONENT_ID,
  'data-garden-version': version,
  className: MenuStyles['c-menu__item__meta']
})`
  ${props => retrieveTheme(COMPONENT_ID, props)};
`;

/** @component */
export default ItemMeta;
