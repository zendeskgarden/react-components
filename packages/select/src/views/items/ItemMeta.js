import { retrieveTheme } from '@zendesk/garden-react-theming';
import { ItemMeta as MenuItemMeta } from '@zendesk/garden-react-menus';

import { version } from '../../../package.json';
const COMPONENT_ID = 'select.item_meta';

/**
 * Accepts all `<span>` props
 */
const ItemMeta = MenuItemMeta.extend.attrs({
  'data-garden-id': COMPONENT_ID,
  'data-garden-version': version
})`
  ${props => retrieveTheme(COMPONENT_ID, props)};
`;

/** @component */
export default ItemMeta;
