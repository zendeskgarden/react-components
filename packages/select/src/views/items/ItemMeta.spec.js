import React from 'react';
import { shallow } from 'enzyme';
import ItemMeta from './ItemMeta';

describe('ItemMeta', () => {
  it('renders default styling', () => {
    const wrapper = shallow(<ItemMeta />);

    expect(wrapper).toMatchSnapshot();
  });
});
