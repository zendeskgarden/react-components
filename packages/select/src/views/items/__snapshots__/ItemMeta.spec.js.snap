// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemMeta renders default styling 1`] = `
<span
  className="ItemMeta-jmIeJN c-menu__item__meta bTnHVY"
  data-garden-id="select.item_meta"
  data-garden-version="version"
/>
`;
